sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -58,6 +58,9 @@ class SparkSqlParser(conf: SQLConf) extends AbstractSqlParser(conf) {
 class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder(conf) {
   import org.apache.spark.sql.catalyst.parser.ParserUtils._
 
+  private def normalizeConfigString(s: String) =
```

Comment:
I checked `tableIdentifier`, it includes `BACKQUOTED_IDENTIFIER`, and in `AstBuilder` we just use `xxx.getText`. Why does `getText` not work here?

Reply:
I looked into `tableIdentifier`, but I couldn't find why back-quotes are removed in the case. I replaced `BACKQUOTED_IDENTIFIER` with `quotedIdentifier` in [this commit](https://github.com/apache/spark/pull/29146/commits/08feee31c5d21aa75f9981efe59f583d46042cb3), then it seems back-quotes are removed by the ANTLR parser. Do you know something about that?

Reply:
I found it. See `PostProcessor.exitQuotedIdentifier`. `quotedIdentifier` is a special parser rule which has post porcessor to remove back-quotes.

Reply:
Oh, nice. I see...